taxi/tests/test_views.py

```diff
@@ -0,0 +1,180 @@
+
+
+class PublicTests(TestCase):
+    def setUp(self) -> None:
+        manufacturer = Manufacturer.objects.create(
+            name="test",
+            country="test country"
+        )
+        Driver.objects.create(
+            username="test",
+            password="test12345"
+        )
+        Car.objects.create(
+            model="test",
+            manufacturer=manufacturer
+        )
+
+    def test_login_required_driver(self) -> None:
+        res = self.client.get(reverse("taxi:driver-list"))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:driver-detail", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:driver-create"))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:driver-delete", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:driver-update", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+
+    def test_login_required_car(self) -> None:
+        res = self.client.get(reverse("taxi:car-list"))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:car-detail", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:car-create"))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:car-delete", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:car-update", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+
+    def test_login_required_manufacturer(self) -> None:
+        res = self.client.get(reverse("taxi:manufacturer-list"))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:manufacturer-create"))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:manufacturer-update", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:manufacturer-delete", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateTests(TestCase):
+    @classmethod
+    def setUpTestData(cls) -> None:
+        Driver.objects.create(
+            username="test1",
+            password="test12345",
+            license_number="test1"
+        )
+        manufacturer = Manufacturer.objects.create(
+            name="test1",
+            country="test country1"
+        )
+        Manufacturer.objects.create(
+            name="test2",
+            country="test country2"
+        )
+        Car.objects.create(
+            model="test",
+            manufacturer=manufacturer
+        )
+
+    def setUp(self) -> None:
+        user = Driver.objects.create_superuser(
+            username="test2",
+            password="test12345",
```

Comment:
The `password` field should be hashed when creating a `Driver` instance. Consider using Django's `set_password` method to ensure the password is stored securely.